snuba/cli/optimize.py

```diff
@@ -39,7 +38,9 @@ def optimize(
     setup_logging(log_level)
 
     dataset = get_dataset(dataset_name)
-    table = enforce_table_writer(dataset).get_schema().get_local_table_name()
+    writable_storage = dataset.get_writable_storage()
+    assert writable_storage is not None
```

Comment:
let's add a message to this assertion
`assert writable_storage is not None, "something on why this is an issue"
Since this would be the error the user would receive.

Also, out of curiosity, what is the reason to stop using enforce_table_writer here ?

Reply:
Will do. I think `enforce_table_writer` is just a shortcut to get the writer and make this assertion. But we need writable storage anyway on 61 now so it doesn't help much.